client/concordclient/include/client/concordclient/concord_client.hpp

```diff
@@ -111,6 +111,15 @@ struct ConcordClientConfig {
   SubscribeConfig subscribe_config;
 };
 
+struct StreamSnapshotRequest {
+  uint64_t snapshot_id;
+  std::string last_received_key;
+};
+struct StreamRequest {
+  // Using a veriant for type subsetting.
+  std::variant<StreamSnapshotRequest> request;
```

Comment:
Could you please explain the rationale for using a variant?

Reply:
ConcordClient is a layer that formalizes the service provided by various clients. It provides an abstraction using Send(), Subscribe(), Unsubscribe() and readStream() APIs. Now let's assume that we have more clients which will provide a subscribe functionality on some endless stream. For all those subsets of types, SubscibeRequest will be used. Similarly, all those clients which provide Stream service (for finite streams) will be provided under the readStream() API. The StreamRequest will represent the subset of all those types. This variant is used for type subsetting of all such future usecases.

Reply:
Can we use function overloading and define `StreamSnapshotRequest` and future types? I think it might be easier to use and understand.

Reply:
Thanks, @dartdart26, let me do this with function overloading. Actually, function overloading has a drawback, that the boilerplate code either needs to be re-written for each type, but using variant type subsetting there will be a single boilerplate code in the dispatcher module and the dispatcher will call the different functions. I am fine with either way as this might reduce the complexity of each new overload.